internal/httphandler/data_request_handler.go

```diff
@@ -90,6 +96,43 @@ func (d *dataRequestHandler) dataQuery(response http.ResponseWriter, request *ht
 	utils.SendHTTPResponse(response, http.StatusOK, data)
 }
 
+func (d *dataRequestHandler) dataRangeQuery(response http.ResponseWriter, request *http.Request) {
+	payload, respondedErr := extractVerifiedQueryPayload(response, request, constants.GetDataRange, d.sigVerifier)
+	if respondedErr {
+		return
+	}
+	query := payload.(*types.GetDataRangeQuery)
+
+	if !d.db.IsDBExists(query.DbName) {
+		utils.SendHTTPResponse(response, http.StatusBadRequest, &types.HttpResponseErr{
+			ErrMsg: "error db '" + query.DbName + "' doesn't exist",
+		})
+		return
+	}
+
+	data, err := d.db.GetDataRange(query.DbName, query.UserId, query.StartKey, query.EndKey)
+	if err != nil {
+		var status int
+
+		switch err.(type) {
+		case *errors.PermissionErr:
+			status = http.StatusForbidden
+		default:
+			status = http.StatusInternalServerError
+		}
```

Comment:
what about the internal BadRequest? 
e.g. when start > end?
i.e. what  would `q.db.GetIterator(dbName, startKey, endKey)` return in that case?

Reply:
It is a valid request and the response would be nil (as followed in many other databases)